src/query/stream/streams.cpp

```diff
@@ -190,26 +191,29 @@ Streams::StreamsMap::iterator Streams::CreateConsumer(StreamsMap &map, const std
 
   auto *memory_resource = utils::NewDeleteResource();
 
-  auto consumer_function =
-      [interpreter_context = interpreter_context_, memory_resource, stream_name,
-       transformation_name = stream_info.common_info.transformation_name, owner = owner,
-       interpreter = std::make_shared<Interpreter>(interpreter_context_),
-       result = mgp_result{nullptr, memory_resource}](const std::vector<typename TStream::Message> &messages) mutable {
-        auto accessor = interpreter_context->db->Access();
-        EventCounter::IncrementCounter(EventCounter::MessagesConsumed, messages.size());
-        CallCustomTransformation(transformation_name, messages, result, accessor, *memory_resource, stream_name);
-
-        DiscardValueResultStream stream;
-
-        spdlog::trace("Start transaction in stream '{}'", stream_name);
-        utils::OnScopeExit cleanup{[&interpreter, &result]() {
-          result.rows.clear();
-          interpreter->Abort();
-        }};
+  auto consumer_function = [interpreter_context = interpreter_context_, memory_resource, stream_name,
+                            transformation_name = stream_info.common_info.transformation_name, owner = owner,
+                            interpreter = std::make_shared<Interpreter>(interpreter_context_),
+                            result = mgp_result{nullptr, memory_resource},
+                            total_retries = interpreter_context_->config.stream_transaction_conflict_retries](
+                               const std::vector<typename TStream::Message> &messages) mutable {
+    auto accessor = interpreter_context->db->Access();
+    EventCounter::IncrementCounter(EventCounter::MessagesConsumed, messages.size());
+    CallCustomTransformation(transformation_name, messages, result, accessor, *memory_resource, stream_name);
+
+    DiscardValueResultStream stream;
+
+    spdlog::trace("Start transaction in stream '{}'", stream_name);
+    utils::OnScopeExit cleanup{[&interpreter, &result]() {
+      result.rows.clear();
+      interpreter->Abort();
+    }};
+
+    const static std::map<std::string, storage::PropertyValue> empty_parameters{};
+    const bool shall_retry = total_retries != 0;
+    for (uint32_t i = 1; i != total_retries && shall_retry; ++i) {
```

Comment:
I'm not sure I follow. If you set it to 0 this shouldn't execute at all.
```suggestion
    const bool total_tries = total_retries + 1;
    for (uint32_t i = 1; i != total_tries; ++i) {
```
Why not something like this?